lib/date.js

```diff
@@ -98,8 +98,17 @@ module.exports = (key, options) => {
         label: label === labelKey ? v.label : label
       });
     }));
+    const conditionalTranslate = (fieldKey) => {
+      let result = req.translate(fieldKey);
+      if (result === fieldKey) {
+        result = null;
+      }
+      return result;
+    };
```

Comment:
Totally makes sense about having it outside of the scope.  Now I see why you have two arguments including the translate function.

Regarding pure functions just been reading about it.  So is it basically that the function can't rely on other external functions or variables?
I also read it shouldn't have side effects and mutate shared things